pyangbind/lib/serialise.py

```
@@ -164,54 +160,271 @@ def default(self, obj, mode="default"):
+    def yname_ns_func(parent_namespace, element, yname):
+        if not element._namespace == parent_namespace:
+            # if the namespace is different, then precede with the module
+            # name as per spec.
+            return "%s:%s" % (element._defining_module, yname)
+        else:
+            return yname
+
+    @staticmethod
+    def generate_element(obj, parent_namespace=None, flt=False, with_defaults=None):
+        ietf_tree_json_func = make_generate_ietf_tree(pybindIETFJSONEncoder.yname_ns_func)
+        return ietf_tree_json_func(obj, parent_namespace=parent_namespace, flt=flt, with_defaults=with_defaults)
+
+
+class EMF(objectify.ElementMaker):
+    """
+    Override ElementMaker to do netconf specific niceties, e.g.
+     - set the namespace
+     - convert underscore to hyphens for tags
+    """
+
+    def __init__(self, namespace=None, nsmap=None):
+        assert namespace or nsmap, "Must set either namespace or nsmap"
+        if namespace:
+            super(EMF, self).__init__(annotate=False, namespace=namespace, nsmap={None: namespace})
+        elif nsmap:
+            super(EMF, self).__init__(annotate=False, namespace=nsmap[None], nsmap=nsmap)
+
+    def __getattribute__(self, tag):
+        return super(EMF, self).__getattr__(tag.replace("_", "-"))
+
+
+class pybindIETFXMLEncoder(object):
+
+    @classmethod
+    def generate_xml_tree(cls, module_name, module_namespace, tree):
+        doc = EMF(namespace=module_namespace)(module_name)
+
+        def aux(parent, root):
+            for k, v in root.items():
+                k, nsmap = k
+                E = EMF(nsmap=dict(nsmap))
+                if isinstance(v, deque):
+                    # TypedList (e.g. leaf-list or union-list)
+                    for i in v:
+                        el = E(k, str(i))
+                        parent.append(el)
+                elif isinstance(v, list):
+                    for i in v:
+                        assert isinstance(i, dict)
```

Please raise a specific error here.